cmd/dp-deployer/main.go

```diff
@@ -6,31 +6,56 @@ import (
+	"github.com/gorilla/mux"
 	"github.com/namsral/flag"
 )
 
 var (
-	consumerQueue      = flag.String("consumer-queue", "", "sqs consumer queue name")
-	consumerQueueURL   = flag.String("consumer-queue-url", "", "sqs queue url")
-	deploymentRoot     = flag.String("deployment-root", "", "root deployment directory")
-	nomadEndpoint      = flag.String("nomad-endpoint", "http://localhost:4646", "nomad client endpoint")
-	nomadTLSSkipVerify = flag.Bool("nomad-tls-skip-verify", false, "skip tls verification of nomad cert")
-	nomadToken         = flag.String("nomad-token", "", "nomad acl token")
-	nomadCACert        = flag.String("nomad-ca-cert", "", "nomad CA cert file")
-	privateKey         = flag.String("private-key", "", "private key used to decrypt secrets")
-	producerQueue      = flag.String("producer-queue", "", "sqs producer queue name")
-	region             = flag.String("aws-default-region", "", "sqs queue region")
-	verificationKey    = flag.String("verification-key", "", "public key for verifying queue messages")
+	consumerQueue              = flag.String("consumer-queue", "", "sqs consumer queue name")
+	consumerQueueURL           = flag.String("consumer-queue-url", "", "sqs queue url")
+	deploymentRoot             = flag.String("deployment-root", "", "root deployment directory")
+	nomadEndpoint              = flag.String("nomad-endpoint", "http://localhost:4646", "nomad client endpoint")
+	nomadTLSSkipVerify         = flag.Bool("nomad-tls-skip-verify", false, "skip tls verification of nomad cert")
+	nomadToken                 = flag.String("nomad-token", "", "nomad acl token")
+	nomadCACert                = flag.String("nomad-ca-cert", "", "nomad CA cert file")
+	privateKey                 = flag.String("private-key", "", "private key used to decrypt secrets")
+	producerQueue              = flag.String("producer-queue", "", "sqs producer queue name")
+	region                     = flag.String("aws-default-region", "", "sqs queue region")
+	verificationKey            = flag.String("verification-key", "", "public key for verifying queue messages")
+	healthcheckInterval        = flag.String("healthcheck-interval", "10s", "time between calling healthcheck endpoints for check subsystems")
+	healthcheckCriticalTimeout = flag.String("healthcheck-critical-timeout", "60s", "time taken for the health changes from warning state to critical due to subsystem check failures")
+	healthcheckPort            = flag.String("healthcheck-port", ":24200", "port for the healthcheck")
+)
+
+var (
+	// BuildTime represents the time in which the service was built
+	BuildTime string
+	// GitCommit represents the commit (SHA-1) hash of the service that is running
+	GitCommit string
+	// Version represents the version of the service that is running
+	Version string
 )
 
 var wg sync.WaitGroup
 
+type healthcheckConfig struct {
+	IntervalStr                string
+	CriticalTimeoutStr         string
+	BindAdd                    string
```

Comment:
```suggestion
	BindAddr                   string
```
"address" should be abbreviated as `Addr`